pypogs/hardware_mounts.py

```diff
@@ -0,0 +1,815 @@
+            formatter = logging.Formatter('%(asctime)s:%(name)s-%(levelname)s: %(message)s')
+            fh.setFormatter(formatter)
+            ch.setFormatter(formatter)
+            self._logger.addHandler(fh)
+            self._logger.addHandler(ch)
+
+        # Start of constructor
+        self._logger.debug('Mount called with model=' + str(model) + ' identity=' + str(identity) + ' name=' \
+                           + str(name) + ' auto_init=' + str(auto_init))
+        self._model = None
+        self._identity = None
+        self._name = 'UnnamedMount'
+        self._is_init = False
+        self._max_speed = (4.0, 4.0) #(alt,azi) degrees/sec
+        self._alt_limit = (-5, 95) #limit degrees
+        self._azi_limit = (None, None) #limit degees
+        self._home_pos = (0, 0) #Home position
+        self._alt_zero = 0 #Amount to subtract from alt.
+        # Only used for model celestron
+        self._cel_serial_port = None
+        # Only used for model ascom
+        self._ascom_scope_alt_axis = 1
+        self._ascom_scope_azi_axis = 0
+        # Thread for rate control
+        self._control_thread = None
+        self._control_thread_stop = True
+        # Cache of the state of the mount
+        self._state_cache = {'alt': 0.0, 'azi': 0.0, 'alt_rate': 0.0, 'azi_rate': 0.0}
+        if name is not None:
+            self.name = name
+        if model is not None:
+            self.model = model
+        if identity is not None:
+            self.identity = identity
+        if model is not None and identity is not None:
+            self.initialize()
+        # Try to get Python to clean up the object properly
+        import atexit, weakref
+        atexit.register(weakref.ref(self.__del__))
+        self._logger.info('Mount instance created with name: ' + self.name + '.')
+
+    def __del__(self):
+        """Destructor, try to stop the mount and disconnect."""
+        try:
+            self._logger.debug('Destructor called, try stop moving and disconnecting')
+        except:
+            pass
+        try:
+            self.stop()
+            self._logger.debug('Stopped')
+        except:
+            pass
+        try:
+            self.deinitialize()
+            self._logger.debug('Deinitialised')
+        except:
+            pass
+        try:
+            self._logger.debug('Destructor finished')
+        except:
+            pass
+
+    @property
+    def debug_folder(self):
+        """pathlib.Path: Get or set the path for debug logging. Will create folder if not
+        existing."""
+        return self._debug_folder
+    @debug_folder.setter
+    def debug_folder(self, path):
+        # Do not do logging in here! This will be called before the logger is set up
+        path = Path(path)
+        if path.is_file():
+            path = path.parent
+        if not path.is_dir():
+            path.mkdir(parents=True)
+        self._debug_folder = path
+
+    @property
+    def state_cache(self):
```

Comment:
Is there a fast method to get the state from ascom? If yes should be implemented here